src/sources/models.js

```diff
@@ -213,6 +224,23 @@ class Subresource {
     throw new Error("validTypes() must be implemented by Subresource subclasses");
   }
   async url(method = 'GET', headers = {}) {
+    /**
+     * Check if the resource has versionInfo, and if it does then we use
+     * the information there rather than constructing the baseName
+    */
+    if (this.resource.versionInfo) {
+      console.log("Subresource::url for type", this.type)
+      if (this.resource.versionInfo[this.type]) {
+        return this.resource.source.urlFor(
+          this.resource.versionInfo[this.type],
+          method, headers
+        )
+      } else {
+        return false;
```

Comment:
Could instead throw a `ResourceKnownNotToExist` error and catch that by consumers (often existence checkers). A common pattern used in the server is to see if something exists by making a HEAD request to the URL, and so we want a way of short-circuiting this where applicable.

Reply:
Yup. Changed to `NotFound`